### PR TITLE
[incubator/drone] Add option to disable DinD

### PR DESCRIPTION
**Pull request by filoozom**
Makes it possible to use the host's Docker instead of DinD for the Drone Agent.

### CODE REVIEW

File: incubator/drone/templates/deployment-agent.yaml [hunk 1]
```diff
@@ -28,21 +28,34 @@ spec:
...
+        - name: DRONE_SERVER
+          value: {{ template "drone.fullname" . }}:9000
+{{- if .Values.dind.enabled }}
+        - name: DOCKER_HOST
+          value: tcp://localhost:2375
+{{- end }}
+        {{ range $key, $value := .Values.agent.env }}
+        - name: {{ $key }}
+          value: {{ $value | quote }}
+        {{ end }}
```
Comment (christian-roggia) [inline incubator/drone/templates/deployment-agent.yaml]: If you keep the original order of the ENV variables it is much clearer which changes you did.

Comment (filoozom) [reply incubator/drone/templates/deployment-agent.yaml]: Right, except indenting back 1 step all env variables to keep spacing consistent throughout the file, I only added the `if` and `end`.

Comment (filoozom) [reply incubator/drone/templates/deployment-agent.yaml]: Actually, it's not consistent in `volumeMounts` either. Guess I should reset the indentation as before and maybe open a different PR to fix styling issues.

**Comment by filoozom** [comment]
@christian-roggia Any reason why you added `appVersion` to `values.yaml`?

**Comment by christian-roggia** [comment]
@filoozom this was a practice I observed in other charts, I decided to follow it since it looks somehow common. If clearer guidelines about which chart values are useful (or required) were available then we might drop `appVersion` altogether.

**Comment by christian-roggia** [comment]
This PR looks good to me @unguiculus @mattfarina

**Comment by christian-roggia** [comment]
@filoozom can you also update the icon to the address https://drone.io/apple-touch-icon.png ? It looks like the SVG is not accepted on websites like kubeapps.

I am asking because the approval/merging process is a little long here and this update is just a minor one.

File: incubator/drone/templates/deployment-agent.yaml [hunk 2]
```diff
@@ -35,14 +35,27 @@ spec:
...
+{{- end }}
           {{ range $key, $value := .Values.agent.env }}
           - name: {{ $key }}
             value: {{ $value | quote }}
           {{ end }}
         resources:
 {{ toYaml .Values.agent.resources | indent 10 }}
+{{- if not .Values.dind.enabled }}
+        securityContext:
+            privileged: true
```
Comment (unguiculus) [inline incubator/drone/templates/deployment-agent.yaml]: Indented too much.

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Comment by filoozom** [comment]
@unguiculus My bad, coppied it from the other place. Fixed both.

**Review by unguiculus — APPROVED** [review]

**Pull request merged**